cloud-media-livestream/keypublisher/README.md

```diff
@@ -0,0 +1,302 @@
+```
+
+Then deploy your function:
+
+```shell
+gcloud functions deploy livestream-key-publisher \
+  --entry-point=keys \
+  --runtime=python310 \
+  --trigger-http \
+  --memory=256MB \
+  --security-level=secure-always \
+  --timeout=60s \
+  --env-vars-file=.env.yaml \
+  --service-account="livestream-cloud-functions@[MY-PROJECT].iam.gserviceaccount.com"
+```
+
+After your function is deployed, the response you get will contain the
+`httpsTrigger.url` field. Save this value for later when you configure your API
+Gateway.
+
+### Create the API and API Gateway
+
+Create the API:
+
+```shell
+gcloud api-gateway apis create livestream-key-publisher-api --display-name="Live Stream Key Publisher API"
+```
+
+Create a file `api-config.yml`. A template file `api-config.template.yml` is
+included here. Copy the template file and replace any needed values. In
+particular `CLOUD_FUNCTION_URL` must be replaced with the URL you received when
+you deployed your function.
+
+Once the file has been created, create your API config:
+
+```shell
+gcloud api-gateway api-configs create livestream-key-publisher-api-config-1 \
+  --display-name="Live Stream Key Publisher API Config v1" \
+  --api=livestream-key-publisher-api \
+  --openapi-spec="api-config.yml" \
+  --backend-auth-service-account="livestream-cf-invoker@[MY-PROJECT].iam.gserviceaccount.com"
+```
+
+Next, create the API Gateway:
+
+```shell
+gcloud api-gateway gateways create livestream-key-publisher-api-gateway \
+  --display-name="Live Stream Key Publisher API Gateway" \
+  --api=livestream-key-publisher-api \
+  --api-config=livestream-key-publisher-api-config-1 \
+  --location=us-central1
+```
+
+Once created, your API must be enabled. Run a command to describe your API:
+
+```shell
+gcloud api-gateway apis describe livestream-key-publisher-api
+```
+
+You will get the `managedService` field in the response. This is the service you
+need to enable. For example,
+
+```shell
+gcloud services enable livestream-key-publisher-api-144h6p0e7bzc1.apigateway.[MY-PROJECT].cloud.goog
+```
+
+Now you need to locate the endpoint URL of your API. Run a command to describe
+your gateway:
+
+```shell
+gcloud api-gateway gateways describe livestream-key-publisher-api-gateway --location=us-central1
+```
+
+You will get the `defaultHostname` field in the response. For example,
+
+```
+defaultHostname: livestream-key-publisher-api-gateway-48mwfke9.uc.gateway.dev
+```
+
+You will send API requests to this URL.
+
+Create an API key to use for making queries to your API, and restrict the API
+key such that only your newly created `livestream-key-publisher-api` can be
+called with the API key. Specify the API restriction with the `managedService`
+field from earlier:
+
+```shell
+gcloud alpha services api-keys create --api-target=service=[managedService]
+```
+
+You will get the `keyString` field in the response. This is your API key. Save
+this key, as you will not be able to retrieve it again.
```

Comment:
This should have a lot of warning signs!!

Reply:
Yes! thank you for the suggestion